source/dynamical_systems/src/IDynamicalSystem.cpp

```diff
@@ -55,7 +55,11 @@ CartesianState IDynamicalSystem<CartesianState>::evaluate(const CartesianState&
     }
     CartesianState result = this->get_base_frame().inverse() * state;
     result = this->compute_dynamics(result);
-    return this->get_base_frame() * result;
+    if (result.is_empty()) {
```

Comment:
I wrote this comment but forgot to press the button to add it to the review... 🤦 

Here a tertiary statement might be preferable. It reduces the repetition of the `return` statement which is a common cause of poor code readability.

`return result.is_empty() ? result : this->get_base_frame() * result;`

Reply:
Much nicer!